lib/impure/volatile.nim

```diff
@@ -0,0 +1,49 @@
+#
+#
+#            Nim's Runtime Library
+#        (c) Copyright 2017 Jeff Ciesielski
+#
+#    See the file "copying.txt", included in this
+#    distribution, for details about the copyright.
+#
+import strutils
+import macros
+
+# Keeps track of the total number of casts in the module (used to
+# generate unique function names used for aliasing)
+var totalCasts {.compiletime.} = 0
+
+const
+  vStoreDecl = """
```

Comment:
I was so focused on emit + macro that I completely missed that possibility.  Good call.